lib/ansible/modules/cloud/amazon/elb_application_lb.py

```diff
@@ -0,0 +1,707 @@
+    elif isinstance(data, collections.Mapping):
+        return dict(map(convert, data.items()))
+    elif isinstance(data, collections.Iterable):
+        return type(data)(map(convert, data))
+    else:
+        return data
+
+
+def convert_tg_name_arn(connection, module, tg_name):
+        
+    try:
+        response = connection.describe_target_groups(Names=[tg_name])
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+    tg_arn =response['TargetGroups'][0]['TargetGroupArn']
+
+    result = tg_arn
+    return result
+
+
+def convert_tg_arn_name(connection, module, tg_arn):
+        
+    try:
+        response = connection.describe_target_groups(TargetGroupArns=[tg_arn])
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+    tg_name = response['TargetGroups'][0]['TargetGroupName']
+
+    result = tg_name
+    return result
+
+
+def wait_for_status(connection, module, elb_arn, status):
+    polling_increment_secs = 15
+    max_retries = (module.params.get('wait_timeout') / polling_increment_secs)
+    status_achieved = False
+
+    for x in range(0, max_retries):
+        try:
+            response = connection.describe_load_balancers(LoadBalancerArns=[elb_arn])
+            if response['LoadBalancers'][0]['State']['Code'] == status:
+                status_achieved = True
+                break
+            else:
+                time.sleep(polling_increment_secs)
+        except botocore.exceptions.ClientError as e:
+            module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+    result = response
+    return status_achieved, result
+
+
+def _compare_tags(current_tags, proposed_tags, purge):
+    """Take a list of current tags and proposed tags and return tags to set and tags to delete. Also takes a purge option."""
+
+    tag_keys_to_delete = []
+    current_tags = boto3_tag_list_to_ansible_dict(current_tags)
+    proposed_tags = boto3_tag_list_to_ansible_dict(proposed_tags)
+    for key in current_tags.keys():
+        if key not in proposed_tags and purge:
+            tag_keys_to_delete.append(key)
+
+    # Remove the keys we're going to delete then compare the dicts to see if modification necessary
+    for key in tag_keys_to_delete:
+        del current_tags[key]
+
+    if current_tags != proposed_tags:
+        tags_need_modify = True
+    else:
+        tags_need_modify = False
+
+    return tag_keys_to_delete, tags_need_modify
+
+
+def _get_subnet_ids_from_subnet_list(subnet_list):
+
+    subnet_id_list = []
+    for subnet in subnet_list:
+        subnet_id_list.append(subnet['SubnetId'])
+
+    return subnet_id_list
+
+
+def get_elb(connection, module):
+
+    try:
+        return connection.describe_load_balancers(Names=[module.params.get("name")])['LoadBalancers'][0]
+    except (ClientError, NoCredentialsError) as e:
```

Comment:
The handling of the exception will be identical as it just uses the message passed from AWS